.travis.yml

```diff
@@ -24,7 +24,13 @@ install:
 script:
   - docker run ${DOCKER_REPOSITORY}:${TRAVIS_COMMIT} /bin/bash -c "./scripts/build.py -t"
   - docker run ${DOCKER_REPOSITORY}:${TRAVIS_COMMIT} /bin/bash -c "./scripts/build.py -t --python-path python3"
+
+after_success:
   - mkdir -p $(dirname ${DOCKER_CACHE_FILE}) ; docker save $(docker history -q ${DOCKER_REPOSITORY}:${TRAVIS_COMMIT} | grep -v '<missing>') | gzip > ${DOCKER_CACHE_FILE};
+  - if [ "$TRAVIS_BRANCH" == "master" ]; then
+    docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD";
```

Comment:
I think Jon may have been concerned specifically about storing login credentials somewhere as this script suggests. I'm certainly concerned by that...

I think this is one case where we need a clear approach to the whole docker workflow. We can't really do this incrementally and hope for the best.